### ISSUE TITLE
"[editMonitoring] not available" when adding new cluster

### ISSUE THREAD

**Issue by Schnitzel** [rancher/terraform-provider-rancher2#525]
We're running into a problem where adding a new cluster via terraform ends up with this error:

```
module.cluster.rancher2_cluster.eks: Creating...

Error: action [editMonitoring] not available on [&{c-jbcvn cluster map[apiServices:https://rancher.example.net/v3/clusters/c-jbcvn/apiservices clusterAlertGroups:https://rancher.example.net/v3/clusters/c-jbcvn/clusteralertgroups clusterAlertRules:https://rancher.example.net/v3/clusters/c-jbcvn/clusteralertrules clusterAlerts:https://rancher.example.net/v3/clusters/c-jbcvn/clusteralerts clusterCatalogs:https://rancher.example.net/v3/clusters/c-jbcvn/clustercatalogs clusterLoggings:https://rancher.example.net/v3/clusters/c-jbcvn/clusterloggings clusterMonitorGraphs:https://rancher.example.net/v3/clusters/c-jbcvn/clustermonitorgraphs clusterRegistrationTokens:https://rancher.example.net/v3/clusters/c-jbcvn/clusterregistrationtokens clusterRoleTemplateBindings:https://rancher.example.net/v3/clusters/c-jbcvn/clusterroletemplatebindings clusterScans:https://rancher.example.net/v3/clusters/c-jbcvn/clusterscans etcdBackups:https://rancher.example.net/v3/clusters/c-jbcvn/etcdbackups namespaces:https://rancher.example.net/v3/clusters/c-jbcvn/namespaces nodePools:https://rancher.example.net/v3/clusters/c-jbcvn/nodepools nodes:https://rancher.example.net/v3/clusters/c-jbcvn/nodes notifiers:https://rancher.example.net/v3/clusters/c-jbcvn/notifiers persistentVolumes:https://rancher.example.net/v3/clusters/c-jbcvn/persistentvolumes projects:https://rancher.example.net/v3/clusters/c-jbcvn/projects remove:https://rancher.example.net/v3/clusters/c-jbcvn self:https://rancher.example.net/v3/clusters/c-jbcvn shell:wss://rancher.example.net/v3/clusters/c-jbcvn?shell=true storageClasses:https://rancher.example.net/v3/clusters/c-jbcvn/storageclasses subscribe:https://rancher.example.net/v3/clusters/c-jbcvn/subscribe templates:https://rancher.example.net/v3/clusters/c-jbcvn/templates tokens:https://rancher.example.net/v3/clusters/c-jbcvn/tokens update:https://rancher.example.net/v3/clusters/c-jbcvn] map[generateKubeconfig:https://rancher.example.net/v3/clusters/c-jbcvn?action=generateKubeconfig importYaml:https://rancher.example.net/v3/clusters/c-jbcvn?action=importYaml viewMonitoring:https://rancher.example.net/v3/clusters/c-jbcvn?action=viewMonitoring]}]

  on modules/cluster/rancher.tf line 1, in resource "rancher2_cluster" "eks":
   1: resource "rancher2_cluster" "eks" {


Releasing state lock. This may take a few moments...
```

our Rancher Terraform files:

```
resource "rancher2_cluster" "eks" {
  name                      = var.cluster_name
  enable_cluster_monitoring = "true"
  cluster_monitoring_input {
    answers = var.cluster_monitoring_answers
  }
}

module "rancher_adopt" {
  source = "./modules/k8s_apply"

  url        = rancher2_cluster.eks.cluster_registration_token[0].manifest_url
  kubeconfig = module.eks.kubeconfig
}

data "rancher2_role_template" "cluster-owner" {
  name = "Cluster Owner"
}

resource "rancher2_cluster_role_template_binding" "example-group-cluster-owner" {
  name               = "example-group-cluster-owner"
  cluster_id         = rancher2_cluster.eks.id
  role_template_id   = data.rancher2_role_template.cluster-owner.id
  group_principal_id = "openldap_group://cn=Cust example,ou=Groups,dc=example,dc=net"
}
```

Interestingly it works though if we first add the cluster without the line ` enable_cluster_monitoring = "true"`, then add the line back to terraform and run a second `terraform apply`.

So it seems there might be an issue with the order that things are added to rancher?

**Comment by rawmind0** [comment]
What provider version are you using??

**Comment by Schnitzel** [comment]
@rawmind0 
```
Terraform v0.13.5
+ provider registry.terraform.io/hashicorp/aws v2.54.0
+ provider registry.terraform.io/hashicorp/http v2.0.0
+ provider registry.terraform.io/hashicorp/kubernetes v1.13.3
+ provider registry.terraform.io/hashicorp/local v2.0.0
+ provider registry.terraform.io/hashicorp/null v3.0.0
+ provider registry.terraform.io/hashicorp/random v3.0.0
+ provider registry.terraform.io/hashicorp/template v2.2.0
+ provider registry.terraform.io/rancher/rancher2 v1.8.3
```

**Comment by rawmind0** [comment]
Are you using provider version v1.8.3 for any reason?? The issue should be solved on currrent provider version, v1.10.6
